server/src/main/java/org/opensearch/index/fielddata/IndexNumericFieldData.java

```diff
@@ -198,24 +198,35 @@ private XFieldComparatorSource comparatorSource(
         MultiValueMode sortMode,
         Nested nested
     ) {
+        XFieldComparatorSource source = null;
```

Comment:
I think this is assigned exactly once in the following switch statement. Can you do the following?
```suggestion
        final XFieldComparatorSource source;
```